src/Sulu/Component/Content/SmartContent/QueryBuilder.php

```diff
@@ -178,17 +178,11 @@ protected function buildOrder($webspaceKey, $locale)
             ? 'DESC' : 'ASC';
 
         $sql2Order = [];
-        $sortBy = $this->getConfig('sortBy', []);
+        $sortBy = $this->getConfig('sortBy');
 
-        if (!empty($sortBy)) {
-            // TODO implement more generic
-            $order = 'page.[i18n:' . $locale . '-' . $sortBy . '] ';
-            if (!in_array($sortBy, ['published', 'created', 'changed', 'authored'])) {
-                $order = sprintf('lower(%s)', $order);
-            }
-
-            $sql2Order[] = $order . ' ' . $sortOrder;
-        } else {
+        if ($sortBy) {
+            $sql2Order[] = 'page.[i18n:' . $locale . '-' . $sortBy . '] ' . $sortOrder;
+        } elseif (!$this->getConfig('includeSubFolders', false)) {
```

Comment:
@chirimoya This fix is kind of a compromise... Instead of Admin sorting there is now a default sorting, which behaves differently based on the `includeSubFolders` flag. If it is not set we sort it after the order of our page tree, and if it is set we don't sort it at all. Then PHPCR seems to return a less confusing order, although it is not respecting the `sulu:order` field.

Reply:
🙈 ... +1